<launch>
  <!--<include file="$(find openni2_launch)/launch/openni2.launch" />-->

  <!--<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video1" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
    <param name="auto_exposure" value="true"/>
    <param name="contrast" value="80"/>
  </node>
  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node>-->

  <param name="cam_w"                     type="int"    value="1280"/>  
  <param name="cam_h"                     type="int"    value="720"/>

  <!--node name="openni2_node" pkg="openni2_launch" type="openni2_launch" /-->

  <remap from="lane_model/lane_model_image" to="app/camera/rgb/image_raw" />
  <!--remap from="camera/ground_image_ipmapped" to="camera/rgb/image_raw" /-->
  <node name="line_detection_node" pkg="line_detection" type="line_detection_node" >
  	
  	<param name="proj_y_start"            	    type="int" 		value="40"/>
  	<param name="proj_image_h"            	    type="int" 		value="80"/>
  	<param name="proj_image_w"            	    type="int" 		value="100"/>
  	<param name="proj_image_horizontal_offset"  type="int"    value="10"/>
    <param name="roi_top_w"            		      type="int" 		value="50"/>
  	<param name="roi_bottom_w"            	    type="int" 		value="50"/>    
  	<param name="detector_size"         	      type="int" 		value="16"/>
    <param name="lane_width"                    type="int"    value="35"/>
  	<param name="path_2features"         	      type="string" value="$(find line_detection)/src/strongClassifiers/classifier_2features.txt"/>
  	<param name="path_30features"         	    type="string" value="$(find line_detection)/src/strongClassifiers/classifier_30features.txt"/>
  </node>

  <param name="camera_name"     type="string"   value="/camera/rgb/image_raw"/>
 
  <param name="f_u"             type="double"   value="896.362732"/>
  <param name="f_v"             type="double"   value="912.085388"/>
  <param name="c_u"             type="double"   value="628.975436"/>
  <param name="c_v"             type="double"   value="405.070026"/>
  <!--param name="f_u"             type="double"   value="624.650635"/>
  <param name="f_v"             type="double"   value="626.987244"/>
  <param name="c_u"             type="double"   value="309.703230"/>
  <param name="c_v"             type="double"   value="231.473613"/-->
  <param name="deg"             type="double"   value="3"/>
  <param name="cam_height"      type="double"   value="18"/>
  <node name="ip_mapper_node" pkg="ip_mapping" type="ip_mapper_node" />

</launch>